### PR TITLE
Docs: What, why and general usage

### PR DESCRIPTION
**Pull request by bstien**
# Why?
We created this library summer of last year, but documentation wasn't a priority at that point. Now that demo/showcase apps are starting to emerge we need to write this to provide info on how to use this.

# What?
- Add docs to types.
- Add what, why and usage docs to `README`.
- `DemoGroupItem`: Remove unused property `groupItemIdentifier`.

### CODE REVIEW

**Comment by bstien** [comment]
After these changes we need to write docs on snapshotting.

File: README.md [hunk 1]
```diff
@@ -1,73 +1,218 @@
...
+
 ```swift
-enum ViewsDemoGroup: String, CaseIterable, DemoGroup, DemoGroupItem {
-    case awesomeView
-    case superbView
+import UIKit
+import DemoKit
+
+// An enum, where all cases refers to a specific demo in your app.
+// Using a
```
Comment (AndreyMomot) [inline README.md]: Seems like an uncompleted phrase

**Review by AndreyMomot — APPROVED** [review]
🔥

File: README.md (hunk 1 above)
Comment (bstien) [reply README.md]: Haha it definitely does😄 I'll see if I can pick up where I left off

**Pull request merged**